Assets/Scripts/Gameplay/ConnectionManagement/ConnectionManager.cs

```diff
@@ -0,0 +1,171 @@
+using System;
```

Comment:
Why is ConnectionManager in Gameplay?

Reply:
cc @pdeschain

Reply:
The goal of this PR was to first refactor it. I intended to move it to a separate assembly in a subsequent PR, but I could do it as part of this one too.